GetDKAN/dkan
# Consolidate metastore retrieve/range methods

Fixes multiple inconsistencies and limitations in the metastore service retrieve methods. Makes it easier to use the metastore service and underlying storage classes to retrieve lists of metadata items or identifiers and either include or exclude unpublished items.

## QA Steps

1. Install demo
2. Install and enable simple_sitemap
3. Run cron. Confirm you see all datasets in /sitemap.xml, including `cedcd327-4e5d-43f9-8eb1-c11850fa7c55`
4. Unpublish the "Florida Bike Lanes" dataset
5. Run cron (or regenerate the sitemap in the admin UI)
6. Confirm `cedcd327-4e5d-43f9-8eb1-c11850fa7c55` is now excluded from /sitemap.xml
7. Confirm `cedcd327-4e5d-43f9-8eb1-c11850fa7c55` is still listed in the Datastore Import Status page as "unpublished"

## Note

* This PR changes several public methods around retrieval, and also changes the signature for the MetastoreStorageInterface::publish method (returns bool instead of string). 
* For this reason, review carefully and make sure you agree with the change, and be sure to make the next release after merge a point, bumping us to (as of this writing) 2.13.
